src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,371 @@
+            matrix_jk[j, k] = expectation_val
+    return matrix_jk
+
+
+def GST(
+    gate_set=Union[tuple, set, list],
+    nshots=int(1e4),
+    noise_model=None,
+    include_empty=False,
+    invert_register=None,
+    Pauli_Liouville=False,
+    backend=None,
+):
+    """This is a wrapper function that runs gate set tomography for a list of gates. One can choose to output the gate set tomography
+    for each gate in the Pauli-Liouville representation or not.
+
+
+    Args:
+        gate_set (tuple, set, list): A list containing :class:`qibo.gates.abstract.Gate`.
+        nshots (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied to simulate noisy computation.
+        include_empty (bool, optional): If ``False``, only perform gate set tomography for the list of gates in ``gate_set``.
+            If ``True``, perform gate set tomography for the list of gates in ``gate_set`` and also empty circuits.
+        invert_register (bool, optional): If ``True``, one needs to specify which qubit(s) to reset.
+        Pauli_Liouville (bool, optional): If ``True``, returns gate set tomography of the gates in the Pauli-Liouville representation.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+    Returns:
+        list(ndarray): List of matrices of the gate(s) in gate set tomography.
+    """
+    matrices = []
+    empty_matrices = []
+    for nqubits in range(1, 3):
+        empty_matrix = _gate_set_tomography(
+            nqubits=nqubits,
+            gate=None,
+            nshots=nshots,
+            noise_model=noise_model,
+            invert_register=invert_register,
+            backend=backend,
+        )
+        empty_matrices.append(empty_matrix)
+        if include_empty:
+            matrices.append(empty_matrix)
+
+    for gate in gate_set:
+        if gate is not None:
+            init_args = signature(gate).parameters
+            if "q" in init_args:
+                nqubits = 1
+            elif "q0" in init_args and "q1" in init_args and "q2" not in init_args:
+                nqubits = 2
+            else:
+                raise_error(
+                    RuntimeError,
+                    f"Gate {gate} is not supported for `GST`, only 1- and 2-qubits gates are supported.",
+                )
+            gate = gate(*range(nqubits))
+
+        if Pauli_Liouville is False:
+            matrices.append(
+                _gate_set_tomography(
+                    nqubits=nqubits,
+                    gate=gate,
+                    nshots=nshots,
+                    noise_model=noise_model,
+                    invert_register=invert_register,
+                    backend=backend,
+                )
+            )
+        elif Pauli_Liouville is True:
+            T = np.array([[1, 1, 1, 1], [0, 0, 1, 0], [0, 0, 0, 1], [1, -1, 0, 0]])
```

Comment:
Just a small comment. The matrix `T` allows for some freedom and can be any invertible matrix. I agree that the choice you have made corresponds to the usual gauge, but I think it would be convenient to expose `T` to the user as an optional argument with the default value you have defined. This would allow for choosing a gauge that optimizes a specific quantity.